test/recipes/controls/aws_parallelcluster_config/network_interfaces_spec.rb

```diff
@@ -0,0 +1,67 @@
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+control 'network_interfaces_configured' do
+  title 'Check that network interfaces have been configured'
+
+  desc 'Check script to configure network interface is created'
+  describe file('/tmp/configure_nw_interface.sh') do
+    it { should exist }
+    its('mode') { should cmp '0644' }
+    its('owner') { should eq 'root' }
+    its('group') { should eq 'root' }
+    its('content') { should match /^# Configure a specific Network Interface according to the OS/ }
+  end
+
+  only_if { !os_properties.virtualized? }
+
+  desc 'Check network interface are configured'
+  device_names = bash("ip -o link | awk '{print substr($2, 1, length($2) -1)}' | grep -v lo").stdout.split(/\n+/)
+  device_names.each do |device_name|
+    if os_properties.debian_family?
+      describe file("/etc/netplan/#{device_name}.yaml") do
+        it { should exist }
+        its('content') { should match /^    #{device_name}:/ }
```

Comment:
Nitpick. Is it critical to have exactly 4 whitespaces at the beginning of the line? Otherwise you could use `\s*` or `\s+`.